### DISCUSSION TITLE
New feature: sending still images

### DISCUSSION THREAD

**Discussion by maciek252** [svpcom/wfb-ng#266] [Ideas]
I'm highly interested in the possibility of sending still images (photos) instead of the video stream. I guess it would need way less bandwidth and be more resistant to temporary network glitches. A photo could be updated at constant periods, say every 10 seconds, and during this time the tx could send the same photo over and over again (or its fragments if it has received an acknowledgment about some part successfully transmitted).

Motivation: I'm mostly flying by waypoints, so this could be a way of checking if everything goes fine, no unexpected masts, power lines etc. The system could switch to video if the available bandwidth would be appropriate.

**Comment by JesterHal** [comment]
This is a very bad idea and makes the flight unsafe. You should always see in real-time where your going and "Return To Home" when you lose video feed. We discourage unsafe flight methods that are based only on GPS navigation. We have seen many times the GPS reporting false data because of glitches. So please fly safe and always remember that safety comes first.

**Comment by maciek252** [comment]
I didn't say that I use only GPS navigation. I'm also using the lidar with a terrain map programmed into the companion computer, so I'm not relaying only on a GPS. If a discrepancy is detected, the drone enters the pre-programmed obstacle-avoiding route. Also, my flights are relatively short, I never fly say 20 km which is ubiquitous in the reports about the video links. In my opinion, this does not provide safety wrt to the collisions with other aircraft even when the video link works perfectly. I fly rather very different missions, say short-distance low altitude terrain-following flights around a mountain and into valleys.

Also, this solution would increase safety for people flying on the basis of the continuous video image, if for some reason the bandwidth becomes too weak for providing a permanent video link in some area. Assuming that there is a flight controller for providing the positional stability of the aircraft, it's better to have a photo than a failed video transmission.

**Comment by JesterHal** [comment]
In my humble opinion no one else is going to use this feature. Everyone else prefers to have stable and robust video link, not still images (1 frame per 10 seconds). You could have a separate link for that through the tunnel link while streaming the regular video. But that's a different topic/request.

**Comment by maciek252** [comment]
Maybe not many users are interested in such an option as the preferred mode, but it would be an alternative for the situations where there is not enough bandwidth for the correct video transmission. And this happens quite often.

**Comment by JesterHal** [comment]
That's not true. For less than 20km range, as you mentioned earlier, you will have with WFB-ng at least 5Mbps bitrate which is more than enough for a crystal clear video stream.

**Reply by maciek252** [reply]
This depends on many factors, such as the wireless activity in the area, the atmospheric conditions (humidity, maybe also sun activity), but mostly on the altitude. Almost all flights reported on youtube are usually well above 200 meters, which I consider not very safe. When you fly 40-60 m. above the ground, the situation is very different and this is true for all the video links I've tried. This includes SIYI, DJI Mavic, Open.hd and Ruby.

**Comment by svpcom** [comment]
@maciek252 wfb-ng is just a transport. It doesn't tied to any video format. Use `rtpjpegpay`  gstreamer element to send still images if you want

**Reply by maciek252** [reply]
OK, thanks for the hint. I'll give it a try. However in the docs it is mentioned that you can download the image and then activate the OSD. Does it mean that the OSD is already provided with the img?

**Reply by JesterHal** [reply]
Yes, type on the GS sudo systemctl enable osd and reboot.
